netsim/roles/__init__.py

```diff
@@ -0,0 +1,26 @@
+#
+# The "node roles" subsystem appends known (statically defined) node roles modules
+# to the topology Plugins (after the user-defined plugins)
+#
+import typing
+from box import Box
+
+from ..data import append_to_list
+
+"""
+Return all nodes matching the specified role(s)
+"""
+def select_nodes_by_role(topology: Box, select: typing.Union[str,list]) -> typing.Generator:
+  roles = select if isinstance(select,list) else [ select ]
+  for node in topology.nodes.values():
+    if node.get('role','router') in select:
```

Comment:
```if node.get('role','router') in roles```

Your variation happens to work due to substring matching, but line 14 suggests a different intent